src/main/java/net/cytonic/cytosis/data/Database.java

```diff
@@ -0,0 +1,106 @@
+import java.util.concurrent.Executors;
+
+public class Database {
+
+    private final ExecutorService worker;
+    private final String host;
+    private final int port;
+    private final String database;
+    private final String username;
+    private final String password;
+    private Connection connection;
+
+    public Database() {
+        this.worker = Executors.newSingleThreadExecutor(Thread.ofVirtual().name("CytosisDatabaseWorker").uncaughtExceptionHandler((t, e) -> Logger.error(STR."An uncaught exception occoured on the thread: \{t.getName()}", e)).factory());
+        this.host = CytosisSettings.DATABASE_HOST;
+        this.port = CytosisSettings.DATABASE_PORT;
+        this.database = CytosisSettings.DATABASE_NAME;
+        this.username = CytosisSettings.DATABASE_USER;
+        this.password = CytosisSettings.DATABASE_PASSWORD;
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            Logger.error("Failed to load database driver");
+            Logger.error(e.toString());
+        }
+    }
+
+    public boolean isConnected() {
+        return (connection != null);
+    }
+
+    public void connect() {
+        worker.submit(() -> {
+            if (!isConnected()) {
+                try {
+                    connection = DriverManager.getConnection(STR."jdbc:mysql://\{host}:\{port}/\{database}?useSSL=false&autoReconnect=true", username, password);
+                } catch (SQLException e) {
+                    Logger.error("Invalid Database Credentials!");
+                    MinecraftServer.stopCleanly();
+                }
+            }
+        });
+
+    }
+
+    public void disconnect() {
+        worker.submit(() -> {
+            if (isConnected()) {
+                try {
+                    connection.close();
+                    Logger.info("Database connection closed!");
+                } catch (SQLException e) {
+                    Logger.error(STR."""
+An error occoured whilst disconnecting from the database. Please report the following stacktrace to Foxikle:\s
+\{Arrays.toString(e.getStackTrace())}""");
+                }
+            }
+        });
+    }
+
+    private Connection getConnection() {
+        return connection;
+    }
+
+    public void createChatTable() {
+        worker.submit(() -> {
+            if (isConnected()) {
+                PreparedStatement ps;
+                try {
+                    ps = getConnection().prepareStatement("CREATE TABLE IF NOT EXISTS cytonicchat (id INT NOT NULL AUTO_INCREMENT, timestamp TIMESTAMP, uuid VARCHAR(36), message TEXT, PRIMARY KEY(id))");
+                    ps.executeUpdate();
+                } catch (SQLException e) {
+                    Logger.error(STR."""
+An error occoured whilst fetching data from the database. Please report the following stacktrace to Foxikle:\s
+\{Arrays.toString(e.getStackTrace())}""");
+                }
+            }
+        });
```

Comment:
I think we want to use completable futures, that way we can keep track of when things conclude.